extension/app/src/lib/utils.ts

```diff
@@ -0,0 +1,62 @@
+export const randomString = (length: number): string => {
+  const charset =
+    "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._~";
+  let result = "";
+  const bytes = new Uint8Array(length);
+
+  crypto.getRandomValues(bytes);
+
+  for (let i = 0; i < length; i++) {
+    result += charset[bytes[i] % charset.length];
+  }
+
+  return result;
+};
+
+export const generatePKCE = async () => {
+  const codeVerifier = randomString(128);
+  const encoder = new TextEncoder();
+  const data = encoder.encode(codeVerifier);
+  const digest = await crypto.subtle.digest("SHA-256", data);
+  const base64Digest = btoa(String.fromCharCode(...new Uint8Array(digest)))
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=+$/, "");
+  return { codeVerifier, codeChallenge: base64Digest };
+};
+
+export const saveAccessToken = (accessToken: string) => {
+  return new Promise((resolve, reject) => {
+    chrome.storage.local.set({ accessToken }, () => {
+      if (chrome.runtime.lastError) {
+        reject(chrome.runtime.lastError);
```

Comment:
`nit`
```suggestion
        return reject(chrome.runtime.lastError);
```

Reply:
You can also create your promise wrapper to do this only once.

Reply:
Not sure why I need the return, and I think we can live without a wrapper so ignoring this one 🙏🏻

Reply:
Explicit returns are so much better, they help read/parse the code more easily 🙃 .